pkg/state/appender.go

```diff
@@ -758,70 +877,14 @@ func (a *txAppender) appendBlock(params *appendBlockParams) error {
 	if err = a.diffStor.saveTxDiff(minerAndRewardDiff); err != nil {
 		return err
 	}
-	blockV5Activated, err := a.stor.features.newestIsActivated(int16(settings.BlockV5))
-	if err != nil {
-		return err
-	}
-	consensusImprovementsActivated, err := a.stor.features.newestIsActivated(int16(settings.ConsensusImprovements))
-	if err != nil {
-		return err
-	}
-	blockRewardDistributionActivated, err := a.stor.features.newestIsActivated(int16(settings.BlockRewardDistribution))
-	if err != nil {
-		return err
-	}
-	invokeExpressionActivated, err := a.stor.features.newestIsActivated(int16(settings.InvokeExpression))
-	if err != nil {
-		return err
-	}
-	// Check and append transactions.
-	var bs proto.BlockSnapshot
-
-	for _, tx := range params.transactions {
-		appendTxArgs := &appendTxParams{
-			chans:                            params.chans,
-			checkerInfo:                      checkerInfo,
-			blockInfo:                        blockInfo,
-			block:                            params.block,
-			acceptFailed:                     blockV5Activated,
-			blockV5Activated:                 blockV5Activated,
-			rideV5Activated:                  rideV5Activated,
-			rideV6Activated:                  rideV6Activated,
-			consensusImprovementsActivated:   consensusImprovementsActivated,
-			blockRewardDistributionActivated: blockRewardDistributionActivated,
-			invokeExpressionActivated:        invokeExpressionActivated,
-			validatingUtx:                    false,
-			currentMinerPK:                   params.block.GeneratorPublicKey,
-		}
-		txSnapshots, errAppendTx := a.appendTx(tx, appendTxArgs)
-		if errAppendTx != nil {
-			return errAppendTx
-		}
-		bs.AppendTxSnapshot(txSnapshots.regular)
-
-		txID, idErr := tx.GetID(a.settings.AddressSchemeCharacter)
-		if idErr != nil {
-			return idErr
-		}
-
-		if len(txSnapshots.regular) == 0 { // sanity check
-			return errors.Errorf("snapshot of txID %q cannot be empty", base58.Encode(txID))
+	if params.snapshot != nil {
```

Comment:
IMHO, I think we could add a parameter `lightNodeMode` to params or to tx.Appender, because a snapshot can be empty if the block is empty